cmd/file_lint.go

```diff
@@ -0,0 +1,165 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+
+	"github.com/daveshanley/vacuum/motor"
+	"github.com/daveshanley/vacuum/rulesets"
+	"github.com/kong/go-apiops/filebasics"
+	"github.com/kong/go-apiops/logbasics"
+	"github.com/spf13/cobra"
+)
+
+var (
+	cmdLintInputFilename string
+	cmdLintInputRuleset  string
+)
+
+func executeLint(cmd *cobra.Command, _ []string) error {
+	severityList := map[string]int{
+		"hint":  0,
+		"info":  1,
+		"warn":  2,
+		"error": 3,
+	}
+
+	verbosity, _ := cmd.Flags().GetInt("verbose")
+	logbasics.Initialize(log.LstdFlags, verbosity)
+
+	inputFilename, err := cmd.Flags().GetString("state")
```

Comment:
As mentioned above, the file name is already available in `cmdLintInputFilename` (same for the other values below)